sandbox_to_csv.py

```diff
@@ -0,0 +1,258 @@
+
+    def write_json_newline(self, recs, fp):
+        with open(fp, 'w') as outfile:
+            for r in recs:
+                outfile.write(json.dumps(r))
+                outfile.write('\n')
+        self.file_names.append(fp)
+
+    def write_csv(self, recs, fp):
+        flat_recs = []
+        for r in recs:
+            flat_recs += self.flattener.process_and_split(r)
+
+        with open(fp, mode='w') as csv_file:
+            field_names = reduce(lambda x, y: set(list(x)+list(y)), flat_recs)
+            writer = csv.DictWriter(csv_file, fieldnames=field_names)
+            writer.writeheader()
+            for flat_rec in flat_recs:
+                writer.writerow(flat_rec)
+        self.file_names.append(fp)
+
+    def write(self, recs, fp):
+        if self.csv:
+            ext = '.csv'
+            self.write_csv(recs, fp+ext)
+        else:
+            ext = '.txt'
+            self.write_json_newline(recs, fp+ext)
+        print('Wrote {} recs to {}'.format(len(recs), fp+ext ))
+
+    def zip_files(self, fp_params):
+        outfp = (self.output_convention+'.zip').format(**fp_params)
+        with zipfile.ZipFile(outfp, 'w') as outzip:
+            for fp in self.file_names:
+                outzip.write(fp, compress_type=zipfile.ZIP_DEFLATED)
+                os.remove(fp)
+        print('Output zip file containing {} files at:\n{}'.format(len(self.file_names), outfp))
+
+
+    def process(self, key):
+        sb,sk = key
+        stream = self.mover.get_data_stream(sb, sk)
+        recs = []
+        for r in self.mover.newline_json_rec_generator(stream):
+            if self.csv:
+                recs += self.flattener.process_and_split(r)
+            else:
+                recs.append(r)
+        self.current_recs += recs
+        return
+
+    def run(self):
+        print('===========START===========')
+        print('Exporting {} {} data between {} and {}'.format(self.pilot, self.message_type, self.sdate, self.edate))
+        t0 = time.time()
+        fp_params = {
+            'pilot': self.pilot,
+            'message_type': self.message_type.lower(),
+            'sdate': self.sdate.strftime('%Y%m%d%H'),
+            'edate': self.edate.strftime('%Y%m%d%H')
+        }
+        fp = lambda filenum: (self.output_convention+'_{filenum}').format(filenum=filenum, **fp_params)
+        sfolder = self.get_folder_prefix(self.sdate)
+        efolder = self.get_folder_prefix(self.edate)
+
+        numkeys = 0
+        filenum = 0
+        numrecs = 0
+        curr_folder = sfolder
+        curr_dt = copy(self.sdate)
+        while curr_folder < efolder:
+            threads = []
+            keys = self.mover.get_fps_from_prefix(self.bucket, curr_folder)
+            if len(keys) > 0:
+                print('Processing {} keys from {}'.format(len(keys), curr_folder))
+            for key in keys:
+                self.process(key)
+                # t = threading.Thread(target = self.process, args=(key,))
```

Comment:
Lets remove the commented code or uncomment it.

Reply:
Removed.